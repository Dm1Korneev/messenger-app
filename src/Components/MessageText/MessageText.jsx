import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

const MessageText = ({ text, isCurrentUserMessage }) => {
  const flexDirection = isCurrentUserMessage ? 'row' : 'row-reverse';

  return (
    <Box display="flex" flex="1" flexDirection={flexDirection} pt={1} px={2}>
      <Typography variant="body2">
        {text}
      </Typography>
    </Box>
  );
};

MessageText.defaultProps = {
  text: '',
};
MessageText.propTypes = {
  text: PropTypes.string,
  isCurrentUserMessage: PropTypes.bool.isRequired,
};

export default MessageText;

