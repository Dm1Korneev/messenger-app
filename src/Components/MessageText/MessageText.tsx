import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';

type Props = {
  text?: string;
  isCurrentUserMessage: boolean;
}

export const MessageText = ({ text = '', isCurrentUserMessage }: Props) => {
  const flexDirection = isCurrentUserMessage ? 'row' : 'row-reverse';

  return (
    <Box display="flex" flex="1" flexDirection={flexDirection} pt={1}>
      <Typography variant="body2">
        {text}
      </Typography>
    </Box>
  );
};
