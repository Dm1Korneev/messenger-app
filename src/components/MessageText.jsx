import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import withStyles from '@material-ui/core/styles/withStyles';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Typography from '@material-ui/core/Typography';

function MessageText(props) {
  const { text, isCurrentUserMessage, classes } = props;

  const ListItemClass = classNames(
    classes.ListItem,
    isCurrentUserMessage && classes.ListItem_reverse,
  );

  return (
    <ListItem className={ListItemClass}>
      <div className={classes.textContent}>
        <ListItemText
          className={classes.ListItemText}
          primary={(
            <Typography component="span" className={classes.ListItemText__text}>
              {text}
            </Typography>
)}
        />
      </div>
    </ListItem>
  );
}

MessageText.defaultProps = {
  text: '',
};
MessageText.propTypes = {
  classes: PropTypes.instanceOf(Object).isRequired,
  text: PropTypes.string,
  isCurrentUserMessage: PropTypes.bool.isRequired,
};

const styles = (theme) => ({
  ListItemText: {
    display: 'flex',
    justifyContent: 'space-between',
    '&:first-child': {
      paddingLeft: theme.spacing.unit * 2,
    },
  },
  ListItem_reverse: {
    flexDirection: 'row-reverse',
  },
  ListItem: {
    alignItems: 'start',
    paddingTop: theme.spacing.unit,
    paddingRight: 0,
    paddingLeft: 0,
    paddingBottom: 0,
  },
  textContent: {
    flexGrow: 1,
  },
  ListItemText__text: {
    fontSize: theme.typography.pxToRem(13),
  },
});

export default withStyles(styles)(MessageText);
