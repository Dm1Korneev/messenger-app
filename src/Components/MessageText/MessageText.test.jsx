import React from 'react';

import MessageText from './MessageText';

jest.mock('@material-ui/core/ListItem', () => global.mockComponent('ListItem'));
jest.mock('@material-ui/core/ListItemText', () => global.mockComponent('ListItemText'));
jest.mock('@material-ui/core/Typography', () => global.mockComponent('Typography'));

const scrollIntoViewMock = jest.fn();
window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

const props = {
  isCurrentUserMessage: false,
};

describe('render MessageText component', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = global.mount(
      <MessageText {...props} />,
    );
  });

  test('component is render', () => {
    expect(wrapper.find('MessageText').length).toBe(1);
  });

  test('ListItem subcomponent is render', () => {
    expect(wrapper.find({ originalcomponent: 'ListItem' }).length).toBe(1);
  });
});

