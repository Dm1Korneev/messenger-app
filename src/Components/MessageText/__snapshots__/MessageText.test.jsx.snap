// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MessageText component snapshot test 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="ListItem"
    test-prop-classname="\\"MessageText-ListItem-3\\""
  >
    <div
      class="MessageText-textContent-4"
    >
      <div
        data-originalcomponent="ListItemText"
        test-prop-classname="\\"MessageText-ListItemText-1\\""
        test-prop-primary="<div data-originalcomponent=\\"Typography\\" test-prop-component=\\"&quot;span&quot;\\" test-prop-classname=\\"&quot;MessageText-ListItemText__text-5&quot;\\" data-reactroot=\\"\\"></div>"
      />
    </div>
  </div>
</DocumentFragment>
`;
